src/librustdoc/clean/types.rs

```diff
@@ -1300,13 +1306,21 @@ crate enum TypeKind {
 }
 
 crate trait GetDefId {
+    /// Doesn't retrieve primitive types `DefId`. Use `def_id_full` if you want it.
     fn def_id(&self) -> Option<DefId>;
+    /// Retrieves all types' `DefId` (including primitives). If you're not interested about
+    /// primitives, use `def_id`.
+    fn def_id_full(&self, cache: &Cache) -> Option<DefId>;
```

Comment:
```suggestion
    /// Retrieves all types' `DefId` (including primitives). If you're not interested about
    /// primitives, use `def_id`.
    fn def_id_full(&self, cache: &Cache) -> Option<DefId>;
    
    fn def_id(&self) -> Option<DefId> {
        self.def_id_full(&Cache::default());
    }
```
That way you don't have to implement the same function twice for each type.

Reply:
`Cache` is a huge type so I'm not sure that creating a new one every time is a good idea. ;)